source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -533,12 +491,134 @@ namespace AdaptiveCards::Rendering::Uwp
+
+                            auto svgDocument = winrt::XmlDocument();
+                            svgDocument.LoadXml(svgText);
+
+                            strongThis->ParseXmlForHeightAndWidth(svgDocument, imageSource);
+                        }
+
+                        // Now that we've parsed the height, we can set the image source
+                        strongThis->SetSvgImageSourceAsync(
+                            imageSource.as<winrt::SvgImageSource>(), stream, uiElement, isAutoSize, parentElement, imageContainer, isVisible);
+                    }
+                });
+        }
+        else
+        {
+            auto setSourceAction = imageSource.as<winrt::BitmapImage>().SetSourceAsync(stream);
+
+            setSourceAction.Completed([weakThis = this->get_weak(), uiElement, isAutoSize, parentElement, imageContainer, isVisible](
+                winrt::IAsyncAction const& /*operation*/, winrt::AsyncStatus status)
+                {
+                    if (status == winrt::AsyncStatus::Completed && isAutoSize)
+                    {
+                        if (auto strongThis = weakThis.get())
+                        {
+                            strongThis->SetAutoSize(uiElement, parentElement, imageContainer, isVisible, false /* imageFiresOpenEvent */);
+                        }
+                    }
+                });
+        }
+    }
+
+    winrt::IAsyncAction render_xaml::XamlBuilder::SetSvgUriSource(winrt::SvgImageSource image, winrt::Uri imageUrl)
+    {
+        try
+        {
+            co_await winrt::resume_foreground(image.Dispatcher());
+            image.as<winrt::SvgImageSource>().UriSource(imageUrl);
+        }
+        catch (winrt::hresult_wrong_thread ex)
+        {
+            // We didn't switch to the UI thread
+        }
+    }
+
+    template<typename T, typename S>
+    winrt::IAsyncAction render_xaml::XamlBuilder::SetSvgImageSourceAsync(winrt::SvgImageSource image,
+                                                                         S stream,
+                                                                         T uiElement,
+                                                                         bool isAutoSize,
+                                                                         winrt::IInspectable parentElement,
+                                                                         winrt::IInspectable imageContainer,
+                                                                         bool isVisible)
+    {
+        try
+        {
+            co_await winrt::resume_foreground(image.Dispatcher());
+            auto setSourceOperation = image.SetSourceAsync(stream);
+
+            setSourceOperation.Completed(
+                [weakThis = this->get_weak(), uiElement, isAutoSize, parentElement, imageContainer, isVisible](
+                    winrt::IAsyncOperation<winrt::SvgImageSourceLoadStatus> const& operation, winrt::AsyncStatus status)
```

Comment:
```suggestion
                   auto operation, auto status)
```
does `auto` type deduction work here?:)

Reply:
Yup this works - I'll add this throughout the file

Reply:
don't forget to use reference where needed:)